marshal/reflect_cache.go

```diff
@@ -0,0 +1,37 @@
+package marshal
+
+import "reflect"
+
+var typeCache = map[reflect.Type]*[]cachedType{}
+
+var nameCache = map[reflect.Type]string{}
+
+type cachedType struct {
+	reflect.StructField
+	JsonTag string
+}
+
+func getTypeFields(structType reflect.Type) *[]cachedType {
+	if res, ok := typeCache[structType]; ok {
+		return res
+	}
+	res := []cachedType{}
+	for i := 0; i < structType.NumField(); i++ {
+		field := structType.Field(i)
+		res = append(res, cachedType{
+			StructField: field,
+			JsonTag:     field.Tag.Get("json"),
+		})
+	}
+	typeCache[structType] = &res
+	return &res
+}
```

Comment:
The type cache can/should contain more information; the helpers in `marshal.go` loop over fields to find one by name. That can instead be a map lookup on some cached struct.

Reply:
While I have a reusable method, we only search for the `forceSendFieldName` field. (I can change the mtehod name/signature). And caching that did not improve performance.

Reply:
Thanks for checking.